packages/story-editor/src/app/rightClickMenu/hooks/useCopyPasteActions.js

```diff
@@ -0,0 +1,238 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useSnackbar } from '@googleforcreators/design-system';
+import { __ } from '@googleforcreators/i18n';
+import { useCallback, useRef, useState } from '@googleforcreators/react';
+import { trackEvent } from '@googleforcreators/tracking';
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+/**
+ * Internal dependencies
+ */
+import { PRESET_TYPES } from '../../../constants';
+import updateProperties from '../../../components/inspector/design/updateProperties';
+import useApplyStyle from '../../../components/panels/design/textStyle/stylePresets/useApplyStyle';
+import getUpdatedSizeAndPosition from '../../../utils/getUpdatedSizeAndPosition';
+import { getTextPresets } from '../../../utils/presetUtils';
+import { useHistory, useStory } from '../..';
+import { getElementStyles } from './utils';
+import { UNDO_HELP_TEXT } from './constants';
+
+/**
+ * Creates the right click menu copy/paste actions.
+ *
+ * @return {Object} Right click menu copy/paste actions
+ */
+const useCopyPasteActions = () => {
+  const undo = useHistory(({ actions }) => actions.undo);
+  const {
+    addAnimations,
+    selectedElement,
+    selectedElementAnimations,
+    updateElementsById,
+  } = useStory(({ state, actions }) => ({
+    addAnimations: actions.addAnimations,
+    selectedElement: state.selectedElements?.[0],
+    selectedElementAnimations: state.selectedElementAnimations,
+    updateElementsById: actions.updateElementsById,
+  }));
+
+  const { showSnackbar } = useSnackbar();
+
+  // Needed to not pass stale refs of `undo` to snackbar
+  const undoRef = useRef(undo);
+  undoRef.current = undo;
+
+  const [copiedElement, setCopiedElement] = useState(null);
+
+  /**
+   * Copy the styles and animations of the selected element.
+   */
+  const handleCopyStyles = useCallback(() => {
+    const oldStyles = { ...copiedElement };
+
+    setCopiedElement({
+      animations: selectedElementAnimations,
+      styles: getElementStyles(selectedElement),
+      type: selectedElement?.type,
+    });
+
+    showSnackbar({
+      actionLabel: __('Undo', 'web-stories'),
+      dismissible: false,
+      message: __('Copied style.', 'web-stories'),
+      onAction: () => {
+        setCopiedElement(oldStyles);
+
+        trackEvent('context_menu_action', {
+          name: 'undo_copy_styles',
+          element: selectedElement?.type,
+          isBackground: selectedElement?.isBackground,
+        });
+      },
+      actionHelpText: UNDO_HELP_TEXT,
+    });
```

Comment:
What are your thoughts on actually moving `copiedElement` to the story provider?

The benefits I see here:
- can create a new story action which doesn't need payload `copySelectedElementStyles()`
- can just leverage our history provider undo action for the snackbar `onAction` here
- `cntrl+z` which is normally associated with undo would now undo copying of element styles and have parity with the possible snackbar action

Reply:
Added this to the story provider state! There's more to cleanup for sure but for now I think it's good enough. We can look at it again in the future copy/paste perf + cleanup ticket.